lib/memoryExtractor.ts

```
@@ -0,0 +1,246 @@
+      lowerMessage.startsWith('why') ||
+      lowerMessage.startsWith('can you') ||
+      lowerMessage.startsWith('could you') ||
+      lowerMessage.startsWith('please')) {
+    return memories;
+  }
+  
+  // Check for pattern matches first (higher priority)
+  let patternMatched = false;
+  for (const pattern of MEMORY_PATTERNS) {
+    const matches = message.match(pattern);
+    if (matches && matches[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)